packages/backend/discovery/xai/arbitrum/diffHistory.md

```diff
@@ -1,3 +1,404 @@
+Generated with discovered.json: 0x9b45f38c2be23002676ce9a7aacd0639ffd49ee3
+
+# Diff at Mon, 15 Apr 2024 09:59:00 GMT:
+
+- author: sekuba (<sekuba@users.noreply.github.com>)
+- comparing to: main@c88e9917cce55305c9e770ab2f0db5a43854522c block: 198866013
+- current block number: 201238253
+
+## Description
+
+### Staking Pools upgrade
+
+This implementation uprade adds support for esXAI staking pools and removes the support for adding to normal staking (Withdrawals and rewards from normal staking remain enabled). Since staking V2 is still disabled and some contracts are still managed by the deployer, a new assessment of admin roles is necessary as soon as staking V2 is enabled.
+
+#### Referee5
+
+The SentryReferee has a new implementation: Referee5.
+This contract has no external `stake()` function so non-pooled (V1) staking is not possible anymore. The new staking entry contract is now `PoolFactory`, which calls the new `onlyPoolFactory`-modified functions inside Referee5. Unstaking of V1 non-pool staked esXAI is still possible with the external `unstake()` function in Referee5.
+
+Referee5 now supports batching of assertions `submitMultipleAssertions()` and rewards `claimMultipleRewards()`.
+
+New mappings are added to assist the assigning of (staking/kyc)-keys to pools, as the number of keys associated with a pool determines the maximum of esXAI that can be staked inside.
+
+#### PoolFactory
+
+The new staking pools get created here (`createPool()`). Pool creators need at least one staking key to create a pool and are able to define a delegate owner address that can post assertions (node duties) on their behalf. The PoolFactory has plenty of pool managing functions (staking/unstaking keys, updating metadata, managing shares), available to the pool creator. 'Shares' of a pool can be allocated by the pool creator within defined bounds to 3 different recipients: 1) The pool creator / owner, 2) Owners of keys staked in the pool, 3) Stakers of esXAI in the pool. This will direct the esXAI rewards to the different recipients. Claiming of rewards is delegated to the pools themselves.
+
+#### StakingPool
+
+Each staking pool is managed in such a contract. There are no pools created yet since V2 staking is still disabled.
+
+#### BucketTracker
+
+Imported by the StakingPools: Tracks the virtual balances of participants in a staking pool. (Needed for calculation and distribution of dividends)
+
+## Watched changes
+
+```diff
+    contract esXai (0x4C749d097832DE2FEcc989ce18fDc5f1BD76700c) {
+    +++ description: None
+      values.getWhitelistCount:
+-        1
++        2
++++ description: Address allowed to initiate esXAI token transfers. Currently assigned to Referee smart contract, in case this changes, the referee contract description should be updated (and the reason investigated).
++++ severity: MEDIUM
+      values.getWhitelistedAddressAtIndex.1:
++        "0xF9E08660223E2dbb1c0b28c82942aB6B5E38b8E5"
```

Comment:
can you look again at this? it appears that >300 addresses are now whitelisted

Reply:
I think we can safely ignore this whitelist. It overrides esXAIs transfer functions to be valid [only if sender OR receiver is whitelisted](https://arbiscan.io/address/0xb38e2edda6c31d9972cac770f3f404cd0d7b55df#code#F16#L136). Due to permissionless staking pools right now [all the pools get whitelisted](https://dashboard.tenderly.co/tx/arbitrum/0x87ffc96eab25a63fd5c661ef8d4779d2fc38669f3b85aaa6a743344a65d76cfe?trace=0.2.0.20).